types/react-signature-canvas/index.d.ts

```diff
@@ -0,0 +1,38 @@
+// Type definitions for react-signature-canvas 1.0
+// Project: https://github.com/agilgur5/react-signature-canvas
+// Definitions by: Kamil Socha <https://github.com/ksocha>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+import * as React from 'react';
+import SignaturePad from 'signature_pad';
+
+export interface SignatureCanvasProps extends SignaturePad.SignaturePadOptions {
+    canvasProps?: React.CanvasHTMLAttributes<HTMLCanvasElement>;
```

Comment:
I believe this is the only proper way to type the attributes of a canvas element. This is how it's used in `React` itself:
```
canvas: React.DetailedHTMLProps<React.CanvasHTMLAttributes<HTMLCanvasElement>, HTMLCanvasElement>;
```

Reply:
Huh, confusing why it would be a generic then 🤔   I guess it passes type-checking in the tests, so that does mean it works o.o.

Reply:
I think it's generic because of consistency. In the React's type definitions, `CanvasHTMLAttributes` is defined as:
```
interface CanvasHTMLAttributes<T> extends HTMLAttributes<T> {
    height?: number | string;
    width?: number | string;
}
```

`IntrinsicElements` interface which defines all html elements looks like:
```
[...]
button: React.DetailedHTMLProps<React.ButtonHTMLAttributes<HTMLButtonElement>, HTMLButtonElement>;
canvas: React.DetailedHTMLProps<React.CanvasHTMLAttributes<HTMLCanvasElement>, HTMLCanvasElement>;
caption: React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
[...]
```

Generic type passed to `HTMLAttributes` is used mostly (if not only) in event handlers definitions, e.g.: `onFocus?: FocusEventHandler<T>;`